packages/library-legacy/src/vote-account.ts

```diff
@@ -234,3 +318,60 @@ function getPriorVoters({buf, idx, isEmpty}: PriorVoters): PriorVoter[] {
     ...buf.slice(0, idx).map(parsePriorVoters),
   ];
 }
+
+/**
+ *
+ * See https://github.com/acheroncrypto/native-to-anchor/blob/master/client/packages/buffer-layout/src/index.ts
+ */
+export function option<T>(
+  layout: BufferLayout.Layout<T>,
+  property?: string
+): BufferLayout.Layout<T | null> {
+  return new OptionLayout<T>(layout, property)
+}
+
+class OptionLayout<T> extends BufferLayout.Layout<T | null> {
```

Comment:
It should also be possible to decode options with tagged unions, right? The discriminator in this case is the `0` or the `1`.

Reply:
I'm sorry, I haven't got this. Can you please point me to some example?

Reply:
This could also be helpful:
https://github.com/solana-labs/solana-program-library/blob/260f80928f796fc78c81efa4dc2a7732665e5a59/token/js/src/serialization.ts#L5-L39